packages/contracts/contracts/optimistic-ethereum/ovm/precompiles/DeployerWhitelist.sol

```diff
@@ -89,9 +103,54 @@ contract DeployerWhitelist {
         address _deployerAddress
     )
         external
-        view
         returns(bool)
     {
-        return allowArbitraryDeployment || whitelistedDeployers[_deployerAddress];
+        bool allowArbitraryDeployment = uint(doOvmSLOAD(ALLOW_ARBITRARY_DEPLOYMENT)) == 1;
+        bool isWhitelistedDeployer = uint(doOvmSLOAD(bytes32(bytes20(_deployerAddress)))) == 1;
+        bool isInitialized = uint(doOvmSLOAD(INITIALIZED_KEY)) != 0;
+        
+        return allowArbitraryDeployment || isWhitelistedDeployer || !isInitialized;
```

Comment:
I feel slight regret that we won't eagerly exit this when we find that `allowArbitraryDeployment == true`, but that's totally fine for our initial version.